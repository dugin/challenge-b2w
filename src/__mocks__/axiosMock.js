import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import {planetMock} from "./responseMock";

export const mock = new MockAdapter(axios);

export const URL = 'https://swapi.co/api/';

mock.onGet(`${URL}planets/`).reply(200, {amount: 50, planet: planetMock, isDone: true});

mock.onGet(`${URL}planets/1/`).reply(200, {data: planetMock});


