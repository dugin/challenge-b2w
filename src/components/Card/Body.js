import React from 'react';
import styled from 'styled-components';
import colors from "../../styles/constants/colors";


const Wrapper = styled.section`  
      padding-bottom: 10px;
      border-bottom: 1px solid ${colors.pattensBlue};
      & > p{
        color: ${colors.baliHai};
        font-size: 14px;
        text-align: start;
      }
`;

const constants = ['POPULATION', 'CLIMATE', 'TERRAIN'];

const Body = (props) => {

    return (
        <Wrapper>
            {constants.map(c=> <p key={c}> {c}: </p>)}
        </Wrapper>
    )
};
export default Body;