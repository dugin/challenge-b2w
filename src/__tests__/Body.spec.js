import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import Body from "../components/Card/Body";
import {planetMock} from "../__mocks__/responseMock";

describe('Card Component', () => {

    const body = shallow(<Body planet={planetMock}/>);
    const constants = ['POPULATION', 'CLIMATE', 'TERRAIN'];


    it('renders properly', () => {
        expect(toJson(body)).toMatchSnapshot();
    });


    it('should contain the array above', () => {
        const texts = body.find('p').map(node => node.text().trim().substring(0, node.text().indexOf(':') - 1));

        constants.forEach(c => expect(texts).toContain(c))
    });
});