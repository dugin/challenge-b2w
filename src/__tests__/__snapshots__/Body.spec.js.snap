// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body Component renders properly 1`] = `
<styled.section>
  <p
    key="POPULATION"
  >
     
    POPULATION
    : 
    <styled.span>
      1000000000000
       
    </styled.span>
  </p>
  <p
    key="CLIMATE"
  >
     
    CLIMATE
    : 
    <styled.span>
      TEMPERATE
       
    </styled.span>
  </p>
  <p
    key="TERRAIN"
  >
     
    TERRAIN
    : 
    <styled.span>
      CITYSCAPE, MOUNTAINS
       
    </styled.span>
  </p>
</styled.section>
`;
